libs/hdf-converters/src/checklist-mapper.ts

```diff
@@ -0,0 +1,454 @@
+
+/**
+ * Tranformer function that splits a string and return array
+ * @param input - string of CCI references
+ * @returns ref - array of CCI references
+ */
+function cciRef(input: string): string[] {
+  return input.split('; ');
+}
+
+const CCI_NIST_MAPPING = new CciNistMapping();
+
+/**
+ * Transformer function that splits string and maps resulting array
+ * into NIST control tags
+ * @param input - string of CCI references
+ * @returns tag - array of NIST Control Tags
+ */
+function nistTag(input: string): string[] {
+  const identifiers: string[] = input.split('; ');
+  return CCI_NIST_MAPPING.nistFilter(
+    identifiers,
+    DEFAULT_STATIC_CODE_ANALYSIS_NIST_TAGS
+  );
+}
+
+/**
+ * Transformer function that checks if the status is 'Not Applicable' returning a 0.
+ * Otherwise, maps severity to IMPACT_MAPPING
+ * @param vuln - checklist vulnerability object
+ * @returns impact - number 0.0, 0.3, 0.5, or 0.7
+ * @default is 0
+ */
+function transformImpact(vuln: Record<string, any>): number {
+  if (vuln.status === 'Not Applicable') return 0.0;
+  const severity = findSeverity(vuln);
+  return IMPACT_MAPPING.get(severity.toString().toLowerCase()) || 0;
+}
+
+/**
+ * Inner function to check is there was a severify override which would alter
+ * the impact of the vulnerability
+ * @param vuln - checklist vulnerability object
+ * @returns - severity
+ */
+function findSeverity(vuln: Record<string, any>): string {
+  if (vuln.severityOverride.length > 0) {
+    return vuln.severityOverride;
+  }
+  return vuln.severity;
+}
+
+/**
+ * Transformer function that uses current heimdall checklist export syntax for
+ * findingDetails attribute to separate a single string into multiple
+ * result objects
+ * @param input - array of one element consisting of {code_desc, status, start_time}
+ * @returns ExecJSON.ControlResult
+ */
+function parseFindingDetails(
+  input: Record<string, any>
+): ExecJSON.ControlResult[] {
+  const results: ExecJSON.ControlResult[] = [];
+  const findings: string = input[0].code_desc;
+  if (findings.length !== 0) {
+    const splitFindings: string[] = findings.split(
+      '--------------------------------\n' // join variable identified from cklResults function in ExportCKLModal.vue
+    );
+    splitFindings.forEach((details) => {
+      let codeDesc = '';
+      let status = '';
+      let message = '';
+      const splitResults: string[] = details.split(/\n(.*)/s, 2); // split using the first new line character only
+      if (
+        splitResults[0] === 'passed' ||
+        splitResults[0] === 'failed' ||
+        splitResults[0] === 'skipped' ||
+        splitResults[0] === 'error'
+      ) {
+        // This does not necessarily work when the code_desc has the word expected
+        // will need to update the export to add a key word that can be used to split code_desc
+        // and message values easier - maybe *results* or *details*
+        const indexOfExpected = splitResults[1].indexOf('expected');
```

Comment:
for now let's assume that everything is status + code_desc.  when we redo the hdf2ckl mapper to make it easier to parse, then we can come back here and handle more exact placement of values instead of message and skip message getting bundled into the code_desc.  it might even be possible to keep track of other values now as well like backtrace, etc.